core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -221,19 +209,16 @@ private void deleteExistingUserToOverwriteIt(String mobileNumber, String overwri
 	}
 
 	@Transactional
-	public UserDTO confirmMobileNumber(UUID userID, String code)
+	public UserDTO confirmMobileNumber(UUID userID, String userProvidedConfirmationCode)
 	{
 		User userEntity = getUserByID(userID);
+		ConfirmationCode confirmationCode = userEntity.getMobileNumberConfirmationCode();
 
-		if (userEntity.getMobileNumberConfirmationCode() == null)
-		{
-			throw MobileNumberConfirmationException.confirmationCodeNotSet(userEntity.getMobileNumber());
-		}
-
-		if (!userEntity.getMobileNumberConfirmationCode().equals(code))
-		{
-			throw MobileNumberConfirmationException.confirmationCodeMismatch(userEntity.getMobileNumber(), code);
-		}
+		verifyConfirmationCode(userEntity, confirmationCode, userProvidedConfirmationCode,
+				() -> MobileNumberConfirmationException.confirmationCodeNotSet(userEntity.getMobileNumber()),
```

Comment:
If we have one generic exception with a placeholder, that would simplify the code I guess. So just a `ConfirmationCodeException` which has a parameter `purposeOfUse` or something like that.